File: memory-domain/src/main/java/com/memory/document/memory/RelationshipInfo.java
package com.memory.document.memory;

import java.util.List;

public record RelationshipInfo(
    List<RelatedMemberInfo> relationships
) {
    public record RelatedMemberInfo(
        Long id,
        String name,
        String nickname,
        String email,
        String profileFileUrl
    ) {}
}

File: memory-domain/src/main/kotlin/com/memory/document/memory/RelationshipInfo.kt
package com.memory.document.memory

class RelationshipInfo(
    private val _relationships: List<RelatedMember>?
) {
    fun relationships(): List<RelatedMember>? = _relationships

    fun firstId(): Long? = relationships()?.firstOrNull()?.id
    fun firstName(): String? = relationships()?.firstOrNull()?.name
    fun firstNickname(): String? = relationships()?.firstOrNull()?.nickname
    fun firstEmail(): String? = relationships()?.firstOrNull()?.email
    fun firstFileUrl(): String? = relationships()?.firstOrNull()?.profileFileUrl
}

class RelatedMember(
    val id: Long?,
    val name: String?,
    val nickname: String?,
    val email: String?,
    val profileFileUrl: String?
)